### ISSUE TITLE
Nunit2.5 Test Output was recently changed to be invalid and NunitXml is go-forward name

### ISSUE THREAD

**Issue by asears** [pester/Pester#1794]
## General summary of the issue
Seems as though Nunit2.5 was recently changed to be disallowed and NunitXml is go-forward test format name.  This was probably long-time coming since Nunit3 is around.

May wish to regress test this change / include the old parameter with deprecation warning until related changes are made to Microsoft repos and some critical external links?  

Here's one place where external dependencies and pipeline examples may break with this change.
https://github.com/PowerShell/vscode-powershell/blob/master/InvokePesterStub.ps1

There seems to be some other changes to the latest release which is preventing tests from running / outputting results.
Will review.

## Describe your environment

Can't run this in an Azure DevOps pipeline due to foreign script link & *.io domain not potentially secure.
Pester 5.1.0

## Expected Behavior

Parameter changes should be deprecated with warnings.

## Current Behavior

Parameter changes broke pipeline.

## Possible Solution? (optional)

Add old parameter or notify clearly on deprecation of critical parameters.

**Comment by asears** [comment]
Seems that $configuration.Run.Exit = $true also was implemented as part of Pester 5.1.  When running invoke-pester twice in a Powershell script for 2 different folders, it would only run one folder unless this was set to $false.

**Comment by fflaten** [comment]
Oh, that's an unexpected edge case.

The option `TestResult.OutputFormat` has been ignored until 5.1.0 since there only was support for NUnit 2.5. The configuration object and documentation used `NUnit2.5` as possible and default value, while the actually `ValidateSet`-attribute on the legacy-parameter `-OutputFormat` accepted `NUnitXml` which we kept.

The configuration object property was never validated as `NUnitXml `was hardcoded in the export-function until now. Using the configuration-object, you actually could have used `$c.TestResult.OutputFormat =  "TheGreatestFormatEver"` in Pester 5.0.4 and still get NUnit2.5 output.

@nohwnd Thoughts? Blame it on bad docs? Add `NUnit2.5` as a valid alias for `NUnitXml`?
"NUnitXml" is still used as format-value in Pester 4.10.1 which is consistent with current behavior and AFAIK the error should only affect users that explicitly set the (now removed) default value in the configuration-object, I might want to blame in on the docs + config default.

I'm assuming any future NUnit3-format would use `NUnit3Xml` unless you had a plan to make NUnit3 primary when it was implemented.

**Comment by fflaten** [comment]
> Seems that $configuration.Run.Exit = $true also was implemented as part of Pester 5.1. When running invoke-pester twice in a Powershell script for 2 different folders, it would only run one folder unless this was set to $false.

That's a separate issue, moved to #1795

**Comment by fflaten** [comment]
If we decide to not implement "NUnit2.5"-value again, we need to raise an issue in vscode-powershell to update or remove it from:
- https://github.com/PowerShell/vscode-powershell/blob/758ea9509c0c81de740ea0e67e1c776a7170d846/InvokePesterStub.ps1#L130-L135
- https://github.com/PowerShell/vscode-powershell/blob/758ea9509c0c81de740ea0e67e1c776a7170d846/InvokePesterStub.ps1#L158-L163

**Comment by nohwnd** [comment]
@fflaten For now I would add `NUnit2.5` as another option and would translate NUnitXML to it as you suggest. Moving forward I am not sure, but I think my original intent was to have the actual versions in the name of the identifier, so NUnit3, NUnit4 etc. 

Not sure if we should make NUnitXml translate to the new format by default, probably not. 

I will add this to 5.1.1xx milestone. PRs should target rel/5.1.1xx branch so we can release this as a patch fix (and the exit issue also). I think I can fix both of those today or tomorrow, unless you are interested in doing that.

**Comment by fflaten** [comment]
@nohwnd My calendar is packed the next couple of days, so if you got time then please do 👍

**Comment by krokofant** [comment]
We've had trouble producing any report with Pester 5.1 that works with the Azure Pipelines [Publish Test Results Task](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/publish-test-results). Not sure how to proceed. Updated to use NUnitXml as OutputFormat but that doesn't seem to do it.

😕

**Comment by nohwnd** [comment]
@krokofant do you see any exceptions? Can you post them?

**Comment by krokofant** [comment]
@nohwnd No exceptions. Just:
> No Result Found to Publish 'C:\agent\_work\1\TestResults\pwsh-mod\pester_pwsh-mod.xml'.
While the xml file looks something like this:
```
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<test-results xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="nunit_schema_2.5.xsd" name="Pester" total="14" errors="0" failures="0" not-run="0" inconclusive="0" ignored="0" skipped="0" invalid="0" date="2020-12-01" time="12:11:07">
  <environment os-version="10.0.19041" cwd="C:\Repo" clr-version="Unknown" machine-name="Foo" user-domain="localhost" user="krokofant" nunit-version="2.5.8.0" platform="Microsoft Windows 10 Enterprise|C:\WINDOWS|\Device\Harddisk0\Partition3" />
  <culture-info current-culture="en-SE" current-uiculture="en-US" />
  <test-suite type="TestFixture" name="Pester" executed="True" result="Success" success="True" time="9.5018" asserts="0" description="Pester">
    <results>
      <test-suite type="TestFixture" name="C:\Repo\posh-mod\Functions\ConfigurationFiles.Tests.ps1" executed="True" result="Success" success="True" time="8.8271" asserts="0" description="C:\Repo\posh-mod\Functions\ConfigurationFiles.Tests.ps1">
        <results>
          <test-suite type="TestFixture" name="ConfigurationFiles" executed="True" result="Success" success="True" time="8.5414" asserts="0" description="ConfigurationFiles">
            <results>
              <test-case description="File paths inside config files should not have double backslashes" name="ConfigurationFiles.File paths inside config files should not have double backslashes" time="1.2332" asserts="0" success="True" result="Success" executed="True" />
            </results>
          </test-suite>
        </results>
      </test-suite>
    </results>
  </test-suite>
</test-results>
```

But my _after coffee_ attempt to use the JUnitXml format worked. So that seems like a reasonable workaround for me.

**Comment by fflaten** [comment]
@krokofant Can't reproduce. Used the following steps for a quick test:
```
steps:
- powershell: |
    Install-Module Pester -RequiredVersion 5.1.0 -Scope CurrentUser -Force
  displayName: 'Install Pester'
- powershell: |
    Import-Module Pester
    Invoke-Pester -Path demo.tests.ps1 -OutputFormat NUnitXml -OutputFile nunit-results.xml
  displayName: 'Run tests'
- task: PublishTestResults@2
  inputs:
    testResultsFormat: 'NUnit'
    testResultsFiles: '**/nunit-*.xml'
    testRunTitle: 'My Test run'
```

Are you sure you PublishTestResults has the correct testResultsFiles-value to match the generated file?
Also tested publishing your xml directly:
![image](https://user-images.githubusercontent.com/3436158/100801086-cba6b080-3427-11eb-9db9-35ed7fa44ac9.png)

**Comment by krokofant** [comment]
@fflaten Thank you for the attention to the issue. It works now with JUnixXml, and I trust your inability to reproduce the issue. It's most likely I missed something.

**Comment by nohwnd** [comment]
Fixed in #1796

**Comment by AlbanBarbosa1982** [comment]
Hi all,

Im running pester 5.1 and $configuration.TestResult.OutputFormat = 'NUnitXML' still produces "nunit_schema_2.5.xsd" name Also I cannot get Junit output to be picked up by azure devops due to xml errors:(
